### DISCUSSION TITLE
Move general asset loads to custom map context

### DISCUSSION THREAD

**Discussion by segabl** [diesel-modding/PAYDAY-2-BeardLib#655] [General]
Regarding https://github.com/simon-wh/PAYDAY-2-BeardLib/commit/bd851e6c8dcd3dbe2e0aea4e737e1d64614bde57, we should probably move these asset loads to custom map loading, I don't see why they need to be loaded in general and just like in this instance, it caused the game to crash on any map instead of just custom maps.

**Comment by Luffyyy** [comment]
There isn't really a reason, just never thought of it and convenience of loading stuff via main.xml and not programmatically.
That being said the unit that got removed, wasn't it used by custom maps? Or did they remove reliance on it

**Comment by Luffyyy** [comment]
Odd, https://github.com/steam-test1/Payday-2-LuaJIT-Complete/blob/8def0fc17f42ca564177023a2423c494d6e74759/core/lib/managers/mission/coreelementunitsequence.lua#L11 implies it's still used so why did they remove it?

**Reply by segabl** [reply]
it seems to be incorrect signatures still since apparently anything thats loaded from db crashes. regardless of that, once it's fixed, should think of loading these things only when a custom map is selected.
